SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -221,7 +239,7 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 
 	if (SpatialSettings->bEnableUnrealLoadBalancer)
 	{
-		ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId()));
+		ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(IntendedVirtualWorkerId));
```

Comment:
If the load balancing strategy returns an invalid virtual worker ID, then the ID set on the component will always be invalid - which wasn't possible before it seems (unless this code could have run before the worker got its own virtual worker ID). Maybe it's worth falling back on the local worker's virtual ID if the LB strategy didn't return a valid ID so the entity doesn't get stuck in limbo?